### PR TITLE
Add theme support

### PR DESCRIPTION
**Pull request by thornbill**

### CODE REVIEW

File: app/src/main/java/org/jellyfin/androidtv/preferences/enums/AppTheme.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package org.jellyfin.androidtv.preferences.enums
+
+enum class AppTheme {
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/preferences/enums/AppTheme.kt]: Can we rename the themes (or the enums) to have consistent naming?

It doesn't make sense to me like this:
- Dark, Theme_Jellyfin
- Emerald, Theme_Jellyfin_Legacy
- Hotdog, Theme_Jellyfin_HotDogStand

File: app/src/main/res/xml/preferences.xml [hunk 2]
```diff
@@ -26,6 +26,11 @@
     </PreferenceCategory>
 
     <PreferenceCategory android:title="@string/pref_general">
+        <ListPreference
+            android:entries="@array/app_theme_entries"
+            android:entryValues="@array/app_theme_values"
+            android:key="app_theme"
+            android:title="App Theme" />
```
Comment (nielsvanvelzen) [inline app/src/main/res/xml/preferences.xml]: Missing `@string`

File: app/src/main/java/org/jellyfin/androidtv/base/AppThemeCallbacks.kt [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+				is PreferencesActivity -> lastPreferencesTheme = it
+				else -> lastTheme = it
+			}
+		}
+	}
+
+	override fun onActivityResumed(activity: Activity) {
+		val lastThemeForActivity = if (activity is PreferencesActivity) lastPreferencesTheme else lastTheme
+		TvApp.getApplication().userPreferences.appTheme.let {
+			if (lastThemeForActivity != null && lastThemeForActivity != it) {
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/base/AppThemeCallbacks.kt]: Isn't it possible to use `activity.setTheme()` here?

Comment (thornbill) [reply app/src/main/java/org/jellyfin/androidtv/base/AppThemeCallbacks.kt]: The activity has to be recreated for the theme to be applied.

File: app/src/main/res/values/array.xml [hunk 4]
```diff
@@ -3,6 +3,14 @@
     xmlns:tools="http://schemas.android.com/tools"
     tools:ignore="MissingTranslation">
 
+    <string-array name="app_theme_entries">
+        <item>@string/pref_theme_dark</item>
+        <item>@string/pref_theme_emerald</item>
+    </string-array>
+    <string-array name="app_theme_values">
```
Comment (nielsvanvelzen) [inline app/src/main/res/values/array.xml]: You forgot to update these with the enum change. (They should be exactly the same as the enums)

File: app/src/main/res/xml/preferences.xml [hunk 5]
```diff
@@ -26,6 +26,12 @@
     </PreferenceCategory>
 
     <PreferenceCategory android:title="@string/pref_general">
+        <ListPreference
+            android:defaultValue="Theme_Jellyfin"
+            android:entries="@array/app_theme_entries"
+            android:entryValues="@array/app_theme_values"
+            android:key="app_theme"
+            android:title="@string/pref_app_theme" />
```
Comment (nielsvanvelzen) [inline app/src/main/res/xml/preferences.xml]: Adding `app:useSimpleSummaryProvider="true"` will show the currently selected theme as "description".

**Pull request merged**